<!DOCTYPE html>
<html lang="en"><head><meta charset="UTF-8"/><meta name="viewport" content="width=device-width, initial-scale=1.0"/><title>Readme Â· GenomicAnnotations.jl</title><link href="https://cdnjs.cloudflare.com/ajax/libs/normalize/4.2.0/normalize.min.css" rel="stylesheet" type="text/css"/><link href="https://fonts.googleapis.com/css?family=Lato|Roboto+Mono" rel="stylesheet" type="text/css"/><link href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" type="text/css"/><link href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/styles/default.min.css" rel="stylesheet" type="text/css"/><script>documenterBaseURL="."</script><script src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.2.0/require.min.js" data-main="assets/documenter.js"></script><script src="siteinfo.js"></script><script src="../versions.js"></script><link href="assets/documenter.css" rel="stylesheet" type="text/css"/></head><body><nav class="toc"><h1>GenomicAnnotations.jl</h1><select id="version-selector" onChange="window.location.href=this.value" style="visibility: hidden"></select><form class="search" id="search-form" action="search/"><input id="search-query" name="q" type="text" placeholder="Search docs"/></form><ul><li class="current"><a class="toctext" href>Readme</a><ul class="internal"><li><a class="toctext" href="#Description-1">Description</a></li><li><a class="toctext" href="#Usage-1">Usage</a></li></ul></li><li><a class="toctext" href="autodocs/">Docstrings</a></li></ul></nav><article id="docs"><header><nav><ul><li><a href>Readme</a></li></ul></nav><hr/><div id="topbar"><span>Readme</span><a class="fa fa-bars" href="#"></a></div></header><h1><a class="nav-anchor" id="GenomicAnnotations.jl-1" href="#GenomicAnnotations.jl-1">GenomicAnnotations.jl</a></h1><h2><a class="nav-anchor" id="Description-1" href="#Description-1">Description</a></h2><p>GenomicAnnotations is a package for reading, modifying, and writing genomic annotations in the GenBank format.</p><h2><a class="nav-anchor" id="Usage-1" href="#Usage-1">Usage</a></h2><p>GenBank files are read with <code>readgbk(gbkfile)</code>, which returns a vector of <code>Chromosome</code>s. If we&#39;re only interested in the first chromosome in <code>example.gbk</code> we only need to store the first element.</p><pre><code class="language-julia">chr = readgbk(&quot;test/example.gbk&quot;)[1]</code></pre><p><code>Chromosome</code>s have five fields, <code>name</code>, <code>header</code>, <code>genes</code>, <code>genedata</code>, and <code>sequence</code>. The <code>name</code> is read from the <code>header</code>, which is stored as a string. The annotation data is stored in <code>genedata</code>, but generally you should use <code>genes</code> to access that data. For example, it can be used to iterate over annotations, and to modify them.</p><pre><code class="language-julia">for gene in chr.genes
    gene.locus_tag = &quot;$(chr.name)_$(gene.locus_tag)&quot;
end

chr.genes[2].locus_tag = &quot;test123&quot;</code></pre><p>Accessing properties that haven&#39;t been stored will return missing. For this reason, it often makes more sense to use <code>get()</code> than to access the property directly.</p><pre><code class="language-julia">if chr.genes[2].pseudo
    println(&quot;Gene 2 is a pseudogene&quot;)
end
# chr.genes[2].pseudo returns missing, so this will throw an error

if get(chr.genes[2], :pseudo, false)
    println(&quot;Gene 2 is a pseudogene&quot;)
end</code></pre><p>The macro <code>@genes</code> can be used to filter through the annotations. The keyword <code>gene</code> is used to refer to the individual <code>Gene</code>s. <code>@genes</code> can also be used to modify annotations.</p><pre><code class="language-julia">@genes(chr, :feature == &quot;CDS&quot;)
@genes(chr, length(gene) &gt; 300)

# The following two expressions are equivalent:
@genes(chr, :feature == &quot;CDS&quot;, length(gene) &gt; 300)
@genes(chr, (:feature == &quot;CDS&quot;) &amp;&amp; (length(gene) &gt; 300))

@genes(chr, :locus_tag == &quot;tag03&quot;)[1].pseudo = true
delete!(@genes(chr, :pseudo)) # Delete all psudogenes
delete!(@genes(chr, length(gene) &lt;= 60)) # Delete all genes 60 nt or shorter</code></pre><p>Gene sequences can be accessed with <code>sequence(gene)</code>. For example, the following code will write the translated sequences of all protein-coding genes to a file:</p><pre><code class="language-julia">using BioSequences
writer = FASTA.Writer(open(&quot;proteins.fasta&quot;, &quot;w&quot;))
for gene in @genes(chr, :feature == &quot;CDS&quot;)
    protseq = translate(convert(RNASequence, sequence(gene)))
    write(writer, FASTA.record(gene.locus_tag, gene.product, protseq))
end
close(writer)</code></pre><p>Genes can be added using <code>addgene!</code>, and <code>sort!</code> can be used to make sure that the resulting annotations are in the correct order.</p><pre><code class="language-julia">newgene = addgene!(chr, &quot;regulatory&quot;, 670:677)
newgene.locus_tag = &quot;reg02&quot;
sort!(chr.genes)</code></pre><p>After modifying the annotations, <code>printgbk(io, chr)</code> can be used to write them to a file.</p><pre><code class="language-julia">open(&quot;updated.gbk&quot;, &quot;w&quot;) do f
    printgbk(f, chr)
end</code></pre><footer><hr/><a class="next" href="autodocs/"><span class="direction">Next</span><span class="title">Docstrings</span></a></footer></article></body></html>
